### PR TITLE
Updated run commands

### PR DESCRIPTION
**Pull request by gibiw**
Fix the API v1 client
--
- removed project code from the client constructor
- changed the auth header name

---

Add a run service
--
- added a client interface
- added create and complete methods

---

Updated run commands
--
- updated the `create` test run command
- added the `complete` test run command
- changed the command structure in the `cmd` folder
- added support CLI and ENV parameters 
- disabled `auth`, `import`, `set` commands

### CODE REVIEW

File: cmd/auth.go [hunk 1]
```diff
@@ -6,9 +6,9 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func init() {
-	rootCmd.AddCommand(authCmd)
-}
+//func init() {
+//	rootCmd.AddCommand(authCmd)
+//}
```
Comment (NickVolynkin) [inline cmd/auth.go:11]: does it stay commented?

Comment (gibiw) [reply cmd/auth.go:11]: I will remove or change it in new versions. I have disabled old commands

File: cmd/root.go [hunk 2]
```diff
@@ -2,42 +2,51 @@ package cmd
...
 )
 
 var rootCmd = &cobra.Command{
-	Use:     "qasectl",
-	Example: "qasectl run create",
-	Run: func(cmd *cobra.Command, args []string) {
-		cmd.Help()
-	},
+	Use:   "qasectl",
+	Short: "This is a CLI tool for Qase TestOps",
```
Comment (NickVolynkin) [inline cmd/root.go:19]:
```suggestion
	Short: "CLI tool for Qase TestOps",
```

File: cmd/run/create/create.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+func Command() *cobra.Command {
+	var (
+		title       string
+		description string
+	)
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   "Create a new test run",
+		Example: "qasectl run create --title 'My test run' --description 'This is a test run' --project 'PRJ' --token 'TOKEN'",
```
Comment (NickVolynkin) [inline cmd/run/create/create.go:27]: Again `qasectl`, not `qli`?

File: go.mod [hunk 4]
```diff
@@ -7,20 +7,36 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/qase-tms/qase-go/qase-api-client v1.0.0
 	github.com/spf13/cobra v1.8.0
+	github.com/spf13/viper v1.18.2
```
Comment (NickVolynkin) [inline go.mod:10]: what's the point of having two `require (... )` blocks?

File: cmd/version/version.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package version
+
+import (
+	"fmt"
+	"github.com/qase-tms/qasectl/internal"
+	"github.com/spf13/cobra"
+)
+
+func VersionCmd() *cobra.Command {
```
Comment (NickVolynkin) [inline cmd/version/version.go:9]: Tried the version command:
```console
$ ./build/qli version
Error: required flag(s) "project", "token" not set
Usage:
  qasectl version [flags]

...

$ ./build/qli version -p a -t b
Qase CLI vv0.1.4-4-g46313a0

File: cmd/run/complete/complete.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	idFlag = "id"
+)
+
+// Command returns a new cobra command for complete runs
+func Command() *cobra.Command {
+	var (
+		runID int64
+	)
+
+	cmd := &cobra.Command{
```
Comment (NickVolynkin) [inline cmd/run/complete/complete.go:22]: Successful command gives no feedback, just blank line. Should be something like
```
Run ${id} completed
```

Using with a run ID that doesn't exist, I get error 404:

```
 ./build/qli run complete -t asdf -p DEMO --id 44                                                
Error: 404 Not Found
Usage:
  qasectl run complete [flags]
  
Examples:
qasectl run complete --id 123 --project 'PRJ' --token 'TOKEN'
...
```

File: cmd/run/create/create.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+// Command returns a new cobra command for create runs
+func Command() *cobra.Command {
+	var (
+		title       string
+		description string
+	)
+
+	cmd := &cobra.Command{
```
Comment (NickVolynkin) [inline cmd/run/create/create.go:24]: Tests

* no token or project: error message about token and project
* no title: `run create` help message
* with token, project and title — ok
* with token, project, title, and description - ok
* same but `--title ''`: Error 400 Bad Request. Not very informative.
* same but `--project ''`: Error: code must have at least 2 elements. I'd say 'characters', but okay.

Comment (NickVolynkin) [reply cmd/run/create/create.go:24]: Overall: nothing blocking, let's roll as it is and improve later.

**Review by NickVolynkin — CHANGES_REQUESTED** [review]
The only blocking problem is that `qli help`, `qli version` and `qli completion` show an error message about token and project being required. All the rest is okay, we can roll as it is now and improve later.

File: go.mod (hunk 4 above)
Comment (gibiw) [reply go.mod:10]: yes.
In the first block, Go stores direct dependencies. 
The second block stores the dependencies on which the direct dependencies depend.

File: cmd/root.go (hunk 2 above)
Comment (gibiw) [reply cmd/root.go]: fixed

File: cmd/run/create/create.go (hunk 3 above)
Comment (gibiw) [reply cmd/run/create/create.go]: fixed

File: cmd/version/version.go (hunk 5 above)
Comment (gibiw) [reply cmd/version/version.go:9]: fixed

File: cmd/run/complete/complete.go (hunk 6 above)
Comment (gibiw) [reply cmd/run/complete/complete.go:22]: Added a message about the successful completion of the run test and if the run test was not found

File: cmd/run/create/create.go (hunk 7 above)
Comment (gibiw) [reply cmd/run/create/create.go:24]: I will think about parameter validation in future versions.

**Pull request merged**